include/albatross/src/tune/finite_difference.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2021 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#ifndef INCLUDE_ALBATROSS_SRC_TUNE_FINITE_DIFFERENCE_HPP_
+#define INCLUDE_ALBATROSS_SRC_TUNE_FINITE_DIFFERENCE_HPP_
+
+namespace albatross {
+
+template <typename Function>
+inline std::vector<double>
+compute_gradient(Function f, const std::vector<double> &params, double f_val,
+                 bool use_async = false) {
+
+  std::vector<std::size_t> inds(params.size());
+  std::iota(std::begin(inds), std::end(inds), 0);
+  const double epsilon = 1e-6;
```

Comment:
Are there concerns over the same eps being used for parameters of all magnitudes? (ie. some parameters may be large like length scales, while others are small like angular length scales or nuggets)

Reply:
Yeah this is one of the downfalls of finite differencing.  In a future iteration I could see us using the prior to derive reasonable steps (kind of doing that already with the `1e-8 * range` logic below).  But basically there's not magic number that will work for all problems.